crates/ruff_linter/src/rules/flake8_type_checking/helpers.rs

```diff
@@ -272,6 +272,11 @@ pub(crate) fn quote_annotation(
     let quote = stylist.quote();
     let mut quote_annotator = QuoteAnnotator::new(semantic, stylist);
     quote_annotator.visit_expr(expr);
+    if quote_annotator.failed {
+        return Err(anyhow::anyhow!(
+            "Annotation already contains quotes that require escaping"
+        ));
+    }
```

Comment:
Can we update the function docs to state that this will return an `Err` in this case?

Additionally, it seems that the function docs is out of date with your previous PR adding support for the cases where it says "(This is currently unsupported.)". We can update that as well.